livemaps-api/ServiceFunctions/FeatureStatePublisher.cs

```diff
@@ -51,25 +50,18 @@ public static class FeatureStatePublisher
             if (prerequisites)
             {
                 await container.CreateIfNotExistsAsync();
-                var bacmapRef = container.GetBlockBlobReference(recentDataFile);
+                var bacmapRef = container.GetBlobClient(recentDataFile);
 
+                BlobDownloadResult result = await bacmapRef.DownloadContentAsync();
+                string deviceStateData = result.Content.ToString();
                 IEnumerable<dynamic> recentdata;
-                using (var ms = new MemoryStream())
-                {
-                    await bacmapRef.DownloadToStreamAsync(ms);
-                    ms.Position = 0;
-                    using (StreamReader reader = new StreamReader(ms, Encoding.UTF8))
-                    {
-                        var bacmapstr = reader.ReadToEnd();
-                        recentdata = JsonConvert.DeserializeObject<IEnumerable<dynamic>>(bacmapstr);
-                    }
-                }
+                recentdata = JsonConvert.DeserializeObject<IEnumerable<dynamic>>(deviceStateData);
 
                 foreach (EventData eventData in events)
                 {
                     try
                     {
-                        string messageBody = Encoding.UTF8.GetString(eventData.Body.Array, eventData.Body.Offset, eventData.Body.Count);
+                        string messageBody = eventData.EventBody.ToString();
```

Comment:
I'm fairly sure that nobody has run this code.    The version of the Event Hubs bindings referenced uses the legacy SDK,  `Microsoft.Azure.EventHubs` which is what the `EventData` model would be passed into the function.  This code is using the `Azure.Messaging.EventHubs` model for `EventData`, which will fail to bind and always be null.

For this to work, you'd either need to reference the `5.0.0-beta` family for the Event Hubs extension package or do a manual conversion from the legacy `EventData` to the current `EventData` in the body of this function.